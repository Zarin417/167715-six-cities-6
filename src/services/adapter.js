export const adaptOffersData = (data) => {
  const adaptedData = ({
    ...data,
    isFavorite: data.is_favorite,
    isPremium: data.is_premium,
    previewImage: data.preview_image,
    maxAdults: data.max_adults,
    host: {
      ...data.host,
      avatarUrl: data.host.avatar_url,
      isPro: data.host.is_pro
    }
  });

  delete adaptedData.is_favorite;
  delete adaptedData.is_premium;
  delete adaptedData.preview_image;
  delete adaptedData.max_adults;
  delete adaptedData.host.avatar_url;
  delete adaptedData.host.is_pro;

  return adaptedData;
};

export const adaptReviewsData = (data) => {
  const adaptedData = {
    ...data,
    user: {
      ...data.user,
      avatarUrl: data.user.avatar_url,
      isPro: data.user.is_pro
    }
  };

  delete adaptedData.user.avatar_url;
  delete adaptedData.user.is_pro;

  return adaptedData;
};
