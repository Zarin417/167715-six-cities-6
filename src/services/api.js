import axios from "axios";

const TIMEOUT = 5000;
const BASE_URL = `https://6.react.pages.academy/six-cities`;

const statusCode = {
  UNAUTHORIZED: 401,
};

export const createAPI = (unAuthorized) => {
  const api = axios.create({
    baseURL: BASE_URL,
    timeout: TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response) => response;

  const onError = (error) => {
    const {response} = error;

    if (response.status === statusCode.UNAUTHORIZED) {
      unAuthorized();
    }

    throw error;
  };

  axios.interceptors.response.use(onSuccess, onError);

  return api;
};
