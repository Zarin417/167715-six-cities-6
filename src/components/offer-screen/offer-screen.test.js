import React from "react";
import {render} from "@testing-library/react";
import configureStore from "redux-mock-store";
import {createMemoryHistory} from "history";
import {Provider} from "react-redux";
import {createAPI} from "../../services/api";
import thunk from "redux-thunk";
import OfferScreen from "./offer-screen";
import {Route, Router} from "react-router";
import TestWrapper from "../../services/test-wrapper/test-wrapper";

const api = createAPI(() => {});
const mockStore = configureStore([thunk.withExtraArgument(api)]);
let history;

describe(`Test 'OfferScreen'`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });

  it(`OfferScreen should render Loader if data is loading`, () => {
    const store = mockStore({
      PLACE_INFO: {isPlaceInfoLoaded: false},
      NEAR_PLACE: {isNearPlacesLoaded: false}
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <OfferScreen />
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`OfferScreen should render correctly`, () => {

    const {container} = render(
        <TestWrapper url={`/offer/11`}>
          <Route
            path={`/offer/:id`}
            exact
            render={() => <OfferScreen />}
          />
        </TestWrapper>
    );

    expect(container).toMatchSnapshot();
  });
});
