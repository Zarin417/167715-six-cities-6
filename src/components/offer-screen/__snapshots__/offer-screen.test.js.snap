// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'OfferScreen' OfferScreen should render Loader if data is loading 1`] = `
<div>
  <div
    class="loader"
  >
    <div
      class="loader__dual-ring"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`Test 'OfferScreen' OfferScreen should render correctly 1`] = `
<div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
                style="display: flex;"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    keks@test.com
                  </span>
                </a>
                <button
                  style="margin-left: 10px;"
                >
                  Log Out
                </button>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--property"
    >
      <section
        class="property"
      >
        <div
          class="property__gallery-container container"
        >
          <div
            class="property__gallery"
          >
            <div
              class="property__image-wrapper"
            >
              <img
                alt="Photo Nice, cozy, warm big bed apartment"
                class="property__image"
                src=""
              />
            </div>
          </div>
        </div>
        <div
          class="property__container container"
        >
          <div
            class="property__wrapper"
          >
            <div
              class="property__mark"
            >
              <span>
                Premium
              </span>
            </div>
            <div
              class="property__name-wrapper"
            >
              <h1
                class="property__name"
              >
                Nice, cozy, warm big bed apartment
              </h1>
              <button
                class="property__bookmark-button
      property__bookmark-button--active button"
                type="button"
              >
                <svg
                  class="property__bookmark-icon"
                  height="33"
                  width="31"
                >
                  <use
                    xlink:href="#icon-bookmark"
                  />
                </svg>
                <span
                  class="visually-hidden"
                >
                  To bookmarks
                </span>
              </button>
            </div>
            <div
              class="property__rating rating"
            >
              <div
                class="property__stars rating__stars"
              >
                <span
                  style="width: 60%;"
                />
                <span
                  class="visually-hidden"
                >
                  Rating
                </span>
              </div>
              <span
                class="property__rating-value rating__value"
              >
                3
              </span>
            </div>
            <ul
              class="property__features"
            >
              <li
                class="property__feature property__feature--entire"
              >
                Room
              </li>
              <li
                class="property__feature property__feature--bedrooms"
              >
                1
                 Bedrooms
              </li>
              <li
                class="property__feature property__feature--adults"
              >
                Max 
                1
                 adults
              </li>
            </ul>
            <div
              class="property__price"
            >
              <b
                class="property__price-value"
              >
                €
                129
              </b>
              <span
                class="property__price-text"
              >
                 night
              </span>
            </div>
            <div
              class="property__inside"
            >
              <h2
                class="property__inside-title"
              >
                What's inside
              </h2>
              <ul
                class="property__inside-list"
              >
                <li
                  class="property__inside-item"
                >
                  Washer
                </li>
              </ul>
            </div>
            <div
              class="property__host"
            >
              <h2
                class="property__host-title"
              >
                Meet the host
              </h2>
              <div
                class="property__host-user user"
              >
                <div
                  class="property__avatar-wrapper user__avatar-wrapper false"
                >
                  <img
                    alt="Host avatar"
                    class="property__avatar user__avatar"
                    height="74"
                    src="img/avatar-angelina.jpg"
                    width="74"
                  />
                </div>
                <span
                  class="property__user-name"
                >
                  Angelina
                </span>
              </div>
              <div
                class="property__description"
              >
                <p
                  class="property__text"
                />
              </div>
            </div>
            <section
              class="property__reviews reviews"
            >
              <h2
                class="reviews__title"
              >
                Reviews ·
                <span
                  class="reviews__amount"
                >
                  0
                </span>
              </h2>
              <form
                action="#"
                class="reviews__form form"
                method="post"
              >
                <label
                  class="reviews__label form__label"
                  for="review"
                >
                  Your review
                </label>
                <div
                  class="reviews__rating-form form__rating"
                >
                  <input
                    class="form__rating-input visually-hidden"
                    id="5-stars"
                    name="rating"
                    type="radio"
                    value="5"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="5-stars"
                    title="perfect"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="4-stars"
                    name="rating"
                    type="radio"
                    value="4"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="4-stars"
                    title="good"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="3-stars"
                    name="rating"
                    type="radio"
                    value="3"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="3-stars"
                    title="not-bad"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="2-stars"
                    name="rating"
                    type="radio"
                    value="2"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="2-stars"
                    title="badly"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                  <input
                    class="form__rating-input visually-hidden"
                    id="1-stars"
                    name="rating"
                    type="radio"
                    value="1"
                  />
                  <label
                    class="reviews__rating-label form__rating-label"
                    for="1-stars"
                    title="terribly"
                  >
                    <svg
                      class="form__star-image"
                      height="33"
                      width="37"
                    >
                      <use
                        xlink:href="#icon-star"
                      />
                    </svg>
                  </label>
                </div>
                <textarea
                  class="reviews__textarea form__textarea"
                  id="comment"
                  maxlength="300"
                  name="comment"
                  placeholder="Tell how was your stay, what you like and what can be improved"
                />
                <div
                  class="reviews__button-wrapper"
                >
                  <p
                    class="reviews__help"
                  >
                    To submit review please make sure to set
                    <span
                      class="reviews__star"
                    >
                      rating
                    </span>
                    and describe your stay with at least
                    <b
                      class="reviews__text-amount"
                    >
                      50 characters
                    </b>
                    .
                  </p>
                  <button
                    class="reviews__submit form__submit button"
                    disabled=""
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </form>
            </section>
          </div>
        </div>
        <section
          class="property__map map"
        >
          <div
            class="leaflet-container leaflet-grab leaflet-touch-drag"
            id="map"
            style="height: 100%; position: relative;"
            tabindex="0"
          >
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer "
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="z-index: 18; left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                      style="width: 256px; height: 256px; left: -130px; top: -72px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              />
              <div
                class="leaflet-pane leaflet-overlay-pane"
              />
              <div
                class="leaflet-pane leaflet-marker-pane"
              >
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="img/pin.svg"
                  style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 12px; top: -204px; z-index: -204;"
                  tabindex="0"
                />
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="img/pin-active.svg"
                  style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -5px; top: 159px; z-index: 159;"
                  tabindex="0"
                />
              </div>
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              />
              <div
                class="leaflet-top leaflet-right"
              />
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                   | © 
                  <a
                    href="https://www.openstreetmap.org/copyright"
                  >
                    OpenStreetMap
                  </a>
                  
          contributors © 
                  <a
                    href="https://carto.com/attributions"
                  >
                    CARTO
                  </a>
                </div>
              </div>
            </div>
          </div>
        </section>
      </section>
      <div
        class="container"
      >
        <section
          class="near-places places"
        >
          <h2
            class="near-places__title"
          >
            Other places in the neighbourhood
          </h2>
          <div
            class="near-places__list places__list"
          >
            <article
              class="near-places__card place-card"
            >
              <div
                class="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                class="near-places__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src=""
                    width="260"
                  />
                </a>
              </div>
              <div
                class=" place-card__info"
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      222
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    class="place-card__bookmark-button
       button"
                    type="button"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 80%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/19"
                  >
                    The good apartment
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  House
                </p>
              </div>
            </article>
          </div>
        </section>
      </div>
    </main>
  </div>
</div>
`;
